WalletWasabi/Http/Models/RequestLine.cs

```diff
@@ -27,11 +29,11 @@ public RequestLine(HttpMethod method, Uri uri, HttpProtocol protocol)
 			StartLineString = Method.Method + SP + URI.AbsolutePath + URI.Query + SP + Protocol + CRLF;
 		}
 
-		public static RequestLine CreateNew(string requestLineString)
+		public static async Task<RequestLine> CreateNewAsync(string requestLineString)
 		{
 			try
 			{
-				var parts = GetParts(requestLineString);
+				var parts = (await GetPartsAsync(requestLineString)).ToArray();
```

Comment:
IMO `GetPartsAsync` could return the list or and array directly.

Reply:
done